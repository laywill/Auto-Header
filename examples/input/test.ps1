#requires -Version 7.0 -Modules Microsoft.PowerShell.Management
using namespace System.IO
using namespace System.Text

<#
.SYNOPSIS
A sample PowerShell script demonstrating key scripting elements
.DESCRIPTION
This script showcases various PowerShell scripting techniques including
parameter handling, custom functions, and system interactions.
#>

param (
    [Parameter(Mandatory=$true)]
    [string]$InputPath,

    [Parameter(Mandatory=$false)]
    [int]$MaxFiles = 10
)

# Custom function to process files
function Process-FileCollection {
    param (
        [Parameter(Mandatory=$true)]
        [string]$Path,

        [Parameter(Mandatory=$false)]
        [int]$Limit = 5
    )

    begin {
        Write-Host "Starting file processing from $Path"
    }

    process {
        try {
            # Get files, limited by the specified max
            $files = Get-ChildItem -Path $Path -File | Select-Object -First $Limit

            foreach ($file in $files) {
                $fileInfo = [FileInfo]::new($file.FullName)

                # Use StringBuilder for efficient string manipulation
                $sb = [StringBuilder]::new()
                [void]$sb.AppendLine("File: $($file.Name)")
                [void]$sb.AppendLine("Size: $($fileInfo.Length) bytes")

                Write-Output $sb.ToString()
            }
        }
        catch {
            Write-Error "Error processing files: $_"
        }
    }

    end {
        Write-Host "File processing completed"
    }
}

# Main script execution
try {
    # Validate input path
    if (-not (Test-Path -Path $InputPath)) {
        throw "Invalid input path: $InputPath"
    }

    # Call the custom function
    $results = Process-FileCollection -Path $InputPath -Limit $MaxFiles

    # Use built-in cmdlet to display results
    $results | Format-Table -AutoSize

    # Additional built-in function example
    Get-Date | Select-Object DateTime, Day, DayOfWeek
}
catch {
    Write-Error "Script execution failed: $_"
    exit 1
}
