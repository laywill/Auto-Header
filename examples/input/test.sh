#!/bin/bash
set -euo pipefail
export LANG=C

# Rest of script...
