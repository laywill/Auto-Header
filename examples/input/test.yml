---
key: value

# Some comment
key2: value2
