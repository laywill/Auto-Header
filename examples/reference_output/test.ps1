#requires -version 5
using namespace System.IO

param(
    [string]$InputPath,
    [string]$OutputPath
)

<# Copyright Example Ltd, UK 2025 #>

# Rest of script...
