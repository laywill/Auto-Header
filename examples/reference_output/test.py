#!/usr/bin/env python3

# Copyright Example Ltd, UK 2025

"""Minimal Python file demonstrating various syntax elements."""

import os  # Standard import
from sys import argv  # Specific import
import math as m  # Import with alias


class Example:
    """A simple class."""

    def __init__(self, value: int):
        """Constructor."""
        self.value = value

    def method(self) -> str:
        """Returns a string."""
        return f"Value: {self.value}"

    @staticmethod
    def static_method():
        """A static method."""
        pass


def function(x: int) -> int:
    """Standalone function."""
    return x * 2


if __name__ == "__main__":
    e = Example(42)  # Instantiate class
    print(e.method())  # Call method
    print(function(5))  # Call function
