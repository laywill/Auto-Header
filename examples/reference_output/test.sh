#!/bin/bash
set -euo pipefail
export LANG=C

# Copyright Example Ltd, UK 2025

# Rest of script...
