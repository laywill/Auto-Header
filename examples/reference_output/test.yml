---
# Copyright Example Ltd, UK 2025

key: value

# Some comment
key2: value2
